config/util/config.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package util
+
+import (
+	"github.com/coreos/go-semver/semver"
+	"github.com/coreos/ignition/v2/config/shared/errors"
+	"github.com/coreos/vcontext/report"
+)
+
+type versionStub struct {
+	Ignition struct {
+		Version string
+	}
+}
+
+// GetConfigVersion parses the version from the given raw config
+func GetConfigVersion(raw []byte) (semver.Version, report.Report, error) {
```

Comment:
It'd be clearer if this commit both 1) added `GetConfigVersion` and 2) switched config.go's `Parse()` function to use it.  That way the second commit isn't doing two things at once.

Reply:
> 2) switched config.go's Parse() function to use it. 

This is applicable only for the new Parse functions, it won't change anything in the existing code

Reply:
`GetConfigVersion` factors out part of the original `Parse`, right?  So it'd be clearer to refactor `Parse` to use `GetConfigVersion`, even though the next commit is going to delete most of the contents of `Parse` anyway.

Reply:
> `GetConfigVersion` factors out part of the original `Parse`, right? So it'd be clearer to refactor `Parse` to use `GetConfigVersion`, even though the next commit is going to delete most of the contents of `Parse` anyway.

Yeah, makes sense